hugolib/site_render.go

```diff
@@ -324,6 +324,7 @@ func (s *Site) renderSitemap() error {
 	page.Sitemap.ChangeFreq = sitemapDefault.ChangeFreq
 	page.Sitemap.Priority = sitemapDefault.Priority
 	page.Sitemap.Filename = sitemapDefault.Filename
+	page.Sitemap.Exclude = sitemapDefault.Exclude
```

Comment:
I'm confused by this.  `page` is the sitemap page itself.  Is it necessary to exclude/include the sitemap from itself?

Actually, this whole function is confusing.  We create a new sitemap page, `n`, on L314, then create another new sitemap page, `page`, on L319.  We make changes to both nodes (L327, L329), then range over `pages` which _shadows_ the `page` variable (not a good practice).  We finally use `n` in the return statement but only use `page.Sitemap.Filename` there.  It appears like the `page` variable here in line 327 could be merged with `n`.  @bep, can you explain what's happening here?  Should we clean this up while we're here?

Reply:
It confused me too. I added this line to follow the convention I saw but I agree, it was confusing. I'll wait to see what bep says.